### PR TITLE
Add ryanbogan as maintainer and codeowner

### PR DESCRIPTION
**Pull request by ryanbogan**
### Description
Adds ryanbogan as maintainer and codeowner for k-NN repository.

 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by ryanbogan** [comment]
@gaiksaya Can you please help with this?

**Comment by gaiksaya** [comment]
> @gaiksaya Can you please help with this?


Waiting for maintainer's approval.

**Review by martin-gaievski — APPROVED** [review]

**Review by naveentatikonda — APPROVED** [review]

**Comment by gaiksaya** [comment]
> > @gaiksaya Can you please help with this?
> 
> Waiting for maintainer's approval.

Added @ryanbogan with maintainer access. Codeowner file is valid now.

**Pull request merged**
